Makefile

```diff
@@ -1,149 +1,152 @@
-# Replace with `make ARCH=xxx` to build for another architecture
 ARCH ?= $(shell flatpak --default-arch)
+GPG_KEY_ID ?= 
 
-# Replace with `make REPO=xxx` to build another repository
+GIT ?= git
+
+BST ?= bst
+BST_ARGS = -o arch $(ARCH) --config build.conf --no-interactive
+
+OSTREE ?= ostree
+
+OUTDIR ?= out
+CACHEDIR ?= cache
 REPO ?= repo
 
-# The branch of the Endless SDK to build
-SDK_BRANCH ?= master
-
-# The version of the Freedesktop runtime we build on
-FDO_RUNTIME_VERSION ?= 1.6
-
-# The version of the GNOME runtime we build on
-GNOME_RUNTIME_VERSION ?= 3.28
-
-BUILD_TAG ?= $(shell date +%Y-%m-%d)
-
-FDO_DEPS = \
-	org.freedesktop.Sdk/${ARCH}/${FDO_RUNTIME_VERSION} \
-	org.freedesktop.Platform/${ARCH}/${FDO_RUNTIME_VERSION} \
-	$()
-
-GNOME_DEPS = \
-	org.gnome.Sdk/${ARCH}/${GNOME_RUNTIME_VERSION} \
-	org.gnome.Sdk.Debug/${ARCH}/${GNOME_RUNTIME_VERSION} \
-	org.gnome.Sdk.Docs/${ARCH}/${GNOME_RUNTIME_VERSION} \
-	org.gnome.Platform/${ARCH}/${GNOME_RUNTIME_VERSION} \
-	$()
-
-LOCALE_DEPS = \
-	org.gnome.Platform.Locale/${ARCH}/${GNOME_RUNTIME_VERSION} \
-	org.gnome.Sdk.Locale/${ARCH}/${GNOME_RUNTIME_VERSION} \
-	$()
-
-SDK_DEPS = \
-	com.endlessm.apps.Sdk.json \
-	com.endlessm.apps.Sdk.appdata.xml \
-	com.endlessm.apps.Platform.appdata.xml \
-	os-release \
-	issue \
-	issue.net \
-	$()
-
-GOOGLE_FONTS = \
-	Fira+Sans \
-	Lato \
-	Libre+Baskerville \
-	Merriweather \
-	Noto+Sans \
-	Noto+Serif \
-	Raleway \
-	Roboto \
-	$()
-
-ICONTHEME_DEPS = \
-	org.freedesktop.Platform.Icontheme.EndlessOS.json \
-	org.freedesktop.Platform.Icontheme.EndlessOS.appdata.xml \
-	$()
-
-# Generic substitution rule
-%: %.in
-	@echo "  GEN   $@" &&                                            \
-	sed -e 's/@@SDK_ARCH@@/${ARCH}/g'                                \
-	    -e 's/@@SDK_BRANCH@@/${SDK_BRANCH}/g'                        \
-	    -e 's/@@GNOME_RUNTIME_VERSION@@/${GNOME_RUNTIME_VERSION}/g'  \
-	    -e 's/@@FDO_RUNTIME_VERSION@@/${FDO_RUNTIME_VERSION}/g'      \
-	    $< > $@.tmp && mv $@.tmp $@
-
-define build-manifest
-	@flatpak-builder --version
-	flatpak-builder \
-		--force-clean --ccache --require-changes \
-		--repo=${REPO} \
-		--arch=${ARCH} \
-		--subject="Build of $1, `date`" \
-		${EXPORT_ARGS} \
-		builddir \
-		$1
+FLATPAK_RUNTIMES_REPO = $(CACHEDIR)/flatpak-runtimes-repo
+FLATPAK_PLATFORM_EXTENSIONS_REPO = $(CACHEDIR)/flatpak-platform-extensions-repo
+
+FLATPAK_REFS = $(shell [ -d "$(REPO)" ] && $(OSTREE) refs --repo $(REPO))
+
+GIT_HOOKS = $(shell [ -d ".git/hooks" ] && echo ".git/hooks/pre-commit")
+
+
+define bundle-ref
+	$(shell flatpak build-bundle $(1) "$(OUTDIR)/$(subst /,-,$(2)).flatpak" $(2))
 endef
 
-all: sdk icontheme
+define clean-ref
+	$(shell rm -f "$(OUTDIR)/$(subst /,-,$(1)).flatpak")
+endef
 
-sdk: ${REPO} $(SDK_DEPS)
-	$(call build-manifest,com.endlessm.apps.Sdk.json)
+define sign-ref
+	$(shell ostree gpg-sign --repo=$(1) $(2) $(GPG_KEY_ID))
+endef
```

Comment:
Rather than gathering the flatpak refs yourself and calling `ostree gpg-sign`, you can just call `flatpak build-sign --gpg-sign $(GPG_KEY_ID) $(REPO)`. If you don't supply a flatpak ID, it will loop through the repo looking for all `app/` and `runtime/` refs to sign.

Reply:
:o that is so much better. Thank you! I changed this to use `flatpak build-sign` and `flatpak build-update-repo` like you demonstrated and removed a bunch of gunk in the process. Indeed, the _sign_ target is unnecessary with that change.